tests/integration/tests/test_version_upgrades.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright 2024 Canonical, Ltd.
+#
+import logging
+from typing import List
+
+import pytest
+from test_util import config, harness, snap, util
+
+LOG = logging.getLogger(__name__)
+
+
+@pytest.mark.node_count(1)
+@pytest.mark.no_setup()
+# @pytest.mark.xfail("cilium failures are blocking this from working")
+@pytest.mark.skipif(
+    not config.VERSION_UPGRADE_CHANNELS, reason="No upgrade channels configured"
+)
+def test_version_upgrades(instances: List[harness.Instance]):
+    channels = config.VERSION_UPGRADE_CHANNELS
+    cp = instances[0]
+
+    if channels[0].lower() == "recent":
+        if len(channels) != 3:
+            pytest.fail(
+                "'recent' requires the number of releases as second argument and the flavour as third argument"
+            )
+        num_channels = int(channels[1])
+        flavour = channels[2]
+        channels = snap.get_latest_channels(num_channels, flavour)
+
+    LOG.info(
+        f"Bootstrap node on {channels[0]} and upgrade through channels: {channels[1:]}"
+    )
+
+    # Setup the k8s snap from the bootstrap channel and setup basic configuration.
+    cp.exec(["snap", "install", "k8s", "--channel", channels[0]])
+    cp.exec(["k8s", "bootstrap"])
```

Comment:
We should also enable all features here. This way this upgrade test also verifies that all our features can update from one snap version to another.